firebase-appdistribution/test-app/src/main/java/com/googletest/firebase/appdistribution/testapp/MainActivity.kt

```diff
@@ -1,23 +1,59 @@
 package com.googletest.firebase.appdistribution.testapp
 
+import android.app.AlertDialog
 import android.content.Intent
+import android.os.Build
 import android.os.Bundle
 import android.util.Log
 import android.view.View
 import android.widget.ProgressBar
 import android.widget.TextView
 import androidx.appcompat.app.AppCompatActivity
 import androidx.appcompat.widget.AppCompatButton
+import com.google.android.gms.tasks.Task
 import com.google.firebase.appdistribution.AppDistributionRelease
 import com.google.firebase.appdistribution.FirebaseAppDistribution
-import com.google.firebase.appdistribution.FirebaseAppDistributionException as FirebaseAppDistributionException1
+import com.google.firebase.appdistribution.UpdateProgress
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
 
 class MainActivity : AppCompatActivity() {
     var firebaseAppDistribution: FirebaseAppDistribution = FirebaseAppDistribution.getInstance()
+    var updateTask: Task<Void>? = null
+    var release: AppDistributionRelease? = null
+    val executorService: ExecutorService = Executors.newFixedThreadPool(1)
+    lateinit var signInButton: AppCompatButton
+    lateinit var signOutButton: AppCompatButton
+    lateinit var checkForUpdateButton: AppCompatButton
+    lateinit var updateAppButton: AppCompatButton
+    lateinit var updateIfNewReleaseAvailableButton: AppCompatButton
+    lateinit var updateIfNewReleaseAvailableButtonBackground: AppCompatButton
+    lateinit var signInButtonBackground: AppCompatButton
```

Comment:
Tried not to do this line multiple times. But if we initialize it outside of onCreate or other function then the app just crashes. Happy to put this back as it was before since I don't think we save any lines of code this way.